### ISSUE TITLE
Bug: SaveSessionTabs() may not be executed in some cases

### ISSUE THREAD

**Issue by IzayoiSake** [files-community/Files#11949]
### Description

When Files is not closed and the computer is restarted directly, Files does not execute SaveSessionTabs(). Therefore, after restarting, the previous page cannot be restored.Even I choose “Continue where you left off”.
![image](https://user-images.githubusercontent.com/96897484/229470272-729092aa-133d-42c3-8831-a88318071940.png)

### Steps To Reproduce

1. Open an instance of Files.
2. Open to a folder.
3. Keeping the FilesAPP not closed, restart the computer.
4. Open Files after computer is turned on.

### Requirements

- eg. Fix crash that occurs when...
- eg. Correct action happens when clicking...

### Files Version

2.4.40.0

### Windows Version

10.0.22621.1485

### Log file

[debug.log](https://github.com/files-community/Files/files/11137435/debug.log)

**Comment by Josh65-2201** [comment]
There have been improvements made in 2.4.60 with one case left which is tracked here #11894

**Comment by Josh65-2201** [comment]
Can you see if this is still an issue in 2.4.60.0

**Comment by IzayoiSake** [comment]
The issue still exists in 2.4.60.0.

**Comment by yaira2** [comment]
Merging with #5993

**Comment by IzayoiSake** [comment]
This problem still exists today, directly shutting down or restarting the computer does not trigger Window_Closed() or AppUnhandledException(). Therefore, SaveSessionTabs() will not be executed.
I implemented this logic through other methods, and the implementation method is as follows:
1.Add a new List<string> CurrentTabsList in GeneralSettingsService.
2.When Tab changes, write Tabs containing ID information into SharedMemory and CurrentTabsList.
3.If ContinueLastSessionOnStartUp is checked, it will detect whether SharedMemory matches CurrentTabsList when starting the Files program.
4.If it does not match, it means that the computer restarted or the Files program was unexpectedly closed without triggering SaveSessionTabs().
5.Restore Tabs recorded in CurrentTabsList.
6. LastSessionTabs has a higher priority than CurrentTabsList, this process will not affect the original logic.
After running it on my computer, no abnormalities have been found for the time being. I want to pull a request for the code, but I am not sure whether it needs approval as written in ACKNOWLEDGE.
Affected functions include: OnNavigatedTo() and SaveSessionTabs() of Class MainPageViewModel.
Added some variables and functions (especially in MainPageViewModel).
